cinema-web/src/main/java/touk/recru/app/facade/Facade.java

```diff
@@ -0,0 +1,16 @@
+package touk.recru.app.facade;
+
+import org.springframework.core.annotation.AliasFor;
+import org.springframework.stereotype.Component;
+
+import java.lang.annotation.*;
+
+@Target({ElementType.TYPE})
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+@Component
+public @interface Facade {
```

Comment:
Tak samo jak w przypadku `UseCase` czym w tym wypadku jest `Facade`? I jaka jest różnica między tymi dwoma adnotacjami?

Reply:
Głównym zadaniem Facade jest walidacja + w pewnych przypadkach (akurat w aplikacji nie udało się pokazać większego zastosowania) może wywoływać różne use-case'y koło siebie. Zwiększa abstrakcję tworząc prosty interfejs do bardziej skomplikowanego sub systemu który wcale nie musi być ze sobą powiązany. 
Adnotacja jest głównie po to aby łatwiej wyszukiwać Fasady w kodzie, jest to zwykły alias z wykorzystaniem Componentu - dependency injection. W przeciwieństwie do UseCase nie musi mieć jednej metody lub paru z overloadowanych, ponieważ może mieć więcej niż jedno zadanie do wykonania.